src/AFQMC/Hamiltonians/HSPotential_Helpers.cpp

```diff
@@ -35,7 +35,7 @@ namespace
 {
 //inline double const& ma::conj(double const& d){return d;}
 //inline float const& ma::conj(float const& f){return f;}
-
+template<class T>
```

Comment:
Since these functions are in the implementation file we need to explicitly instantiate them.

William's branch has a couple macros to make this easier.

Reply:
I feel the template parameter should be bool instead of T.

Reply:
If we go that route, make it an enum or enum class. Names would be clearer than `true` and `false` and if in the future we want to add more template options (mixed precision?) there is less to change.

Reply:
Good point @quantumsteve . We will  want to add mixed precision support in future to do away with QMC_MIXED_PRECISION. If we can pick a pattern that make both r/c and sp/dp combinations simpler to support, we should.

Reply:
There's another problem here. Something that was overlooked here is that the type of the `SpVType_shm_csr_matrix` function parameter for these functions would need to become a template parameter since it is dependent on the `SPValueType`. Following that we could use the `CSR_matrix_type::value_type` to determine if we're dealing with complex or not.

To Ye's suggestion then, we would still need to make explicit instantiations for just the two sp value types (where the rest of the specialization of `csr_matrix` remains the same: `int, std::size_t, shared_allocator`).

Reply:
Does anyone feel like they could write some documentation for this function?  Anyway I make a suggestion for changing the matrix type aliases in another comment which are pretty disgusting anyway which will make instantiating the two template cases much less awkward.